In [1]:
import math

<b>Problem Statement 1:<br>
Blood glucose levels for obese patients have a mean of 100 with a standard deviation of 15.
A researcher thinks that a diet high in raw cornstarch will have a positive effect on
blood glucose levels. A sample of 36 patients who have tried the raw cornstarch diet
have a mean glucose level of 108. Test the hypothesis that the raw cornstarch had an
effect or not.</b>

In [2]:
mu1= 100
sd= 15
n=36
mu2=108

Step-1: State the hypotheses. The population mean is 100.<br>
H0:μ=100 ==> null hypothesis<br>
H1:≠100  ==> Alternate hypothesis<br>

Step-2:Set up the significance level. It is not given in the problem so let’s assume it as 5% (0.05).

This 5% is called Significance Level also known as alpha level (symbolized as α). <br>
It means that if random chance probability is less than 5% then we can conclude that there is difference in behavior of two different population.<br> 
(1- Significance level) is also known as Confidence Level 
i.e. we can say that I am 95% confident that it is not driven by randomness.


Step-3 : Claculate Z Score

In [3]:
z= (mu2-mu1)/ (sd/ math.sqrt(n))
print("Z score = ",z)

Z score =  3.2


by looking at z- table and p-value associated with 3.20 is 0.9993 <br>
The probability of having value less than 108 is 0.9993 and more than or equals to 108 is (1-0.9993)=0.0007.

Step-4: Sice the probability of having mean glucose level more than or equals to 108  is 0.0007 which is  less than 0.05 
so we will reject the Null hypothesis i.e. there is raw cornstarch effect.

<b>Problem Statement 2:<br>
In one state, 52% of the voters are Republicans, and 48% are Democrats. In a second
state, 47% of the voters are Republicans, and 53% are Democrats. Suppose a simple
random sample of 100 voters are surveyed from each state.
What is the probability that the survey will show a greater percentage of Republican
voters in the second state than in the first state?</b>

In [4]:
n1=100
n2=100
p1= 0.52
q1= 0.48
p2= 0.47
q2= 0.53

In [5]:
mu= p1-p2
print("mu: ",mu)

mu:  0.050000000000000044


In [6]:
sd = math.sqrt(((p1 * q1 ) / n1) + ((p2 * q2) /n2))
print("SD: ",sd)

SD:  0.07061869440877536


In [7]:
x=0

In [8]:
z = ( x - mu)/sd

In [9]:
print("Z score: ", z)

Z score:  -0.7080278164104213


From Z table we find that the probability of a z-score being -0.7082 or less is 0.24.

Therefore, the probability that the survey will show a greater percentage of Republican voters 
in the second state than in the first state is 0.24.

<b>Problem Statement 3:<br>
You take the SAT and score 1100. The mean score for the SAT is 1026 and the standard
deviation is 209. How well did you score on the test compared to the average test taker?</b>

In [10]:
x = 1100 
mu = 1026 
sd = 209

In [11]:
z= (x-mu)/sd

In [12]:
z

0.35406698564593303

The ztable gives pvalue as 0.354

In [13]:
print("Score is in the range {} - {}  with a  zscore 0.35".format(mu - sd,mu + sd))

Score is in the range 817 - 1235  with a  zscore 0.35
